Src/CSharpier/SyntaxNodeJsonWriter.cs

```diff
@@ -18,7 +18,7 @@ internal partial class SyntaxNodeJsonWriter
     {
         if (!string.IsNullOrEmpty(value))
         {
-            return $"\"{name}\":{JsonConvert.ToString(value, '"', StringEscapeHandling.Default)}";
+            return $"\"{name}\":\"{JsonEncodedText.Encode(value)}\"";
```

Comment:
Is there a particular reason you're doing this manually? If the goal is to filter some properties, why not make your generators just add things to a dictionary and serialize that?